File: fe/src/app/dto/presenza.dto.ts
export interface Presenza {
    id_presenza: number;
    data: Date;
    ingresso: string; 
    uscita: string;
    totale: string;
    permesso: string;
    ferie: string;
    progetto: Progetto;
  }

export interface Progetto{
    id_progetto: number;
    nome_cliente: string;
    titolo: string;
}

export class PresenzaClass{
    idUtente: number;
    data: Date;
    ingresso: string; 
    uscita: string;
    totale: number;
    permesso: number;
    ferie: number;
    idProgetto: number;
}

File: central-be/src/main/java/it/company/portale/dto/PresenzaDTO.java
package it.valueson.portale.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PresenzaDTO {
    private Long idPresenza;
    private Long idUtente;
    private LocalDate data;
    private LocalTime ingresso;
    private LocalTime uscita;
    private double totale;
    private Integer permesso;
    private Integer ferie;
    private Long idProgetto;
}
